src/cgalutils-applyops.cc

```diff
@@ -34,8 +34,43 @@
 #include <queue>
 #include <unordered_set>
 
+#include <atomic>
+
 namespace CGALUtils {
 
+	// manage changes to CGAL::set_error_behaviour:
+	//  a mutex to guard access
+	std::atomic_flag lockedErrorsLock = ATOMIC_FLAG_INIT;
+	//  the number of times lockErrors has been called (without corresponding unlockErrors)
+	size_t lockedErrorsCount = 0;
+	//  the original error state
+	CGAL::Failure_behaviour lockedErrors = CGAL::Failure_behaviour::ABORT;
+
+#define SPIN_LOCK(flag) do { } while(flag.test_and_set())
+
+	// increments lockedErrorsCount and calls CGAL::set_error_behavior if this is the first [unlocked] call
+	void lockErrors(CGAL::Failure_behaviour behavior)
```

Comment:
I was about to suggest SPINLOCK should call [yield()](http://www.boost.org/doc/libs/1_63_0/doc/html/thread/thread_management.html#thread.thread_management.this_thread.yield), however boost's spinlock does it already, 
```    
void lock()
    {
        for( unsigned k = 0; !try_lock(); ++k )
        {
            boost::detail::yield( k );
        }
    }
```